src/main/java/com/google/songgame/servlets/ChatServlet.java

```diff
@@ -122,37 +123,49 @@ private String getUserId(HttpServletRequest request) throws IOException {
   }
 
   private String getUsername(String userId) {
-    Filter userIdFilter =
-        new FilterPredicate("userId", FilterOperator.EQUAL, userId);
+    Filter userIdFilter = new FilterPredicate("userId", FilterOperator.EQUAL, userId);
     Query userQuery = new Query("User").setFilter(userIdFilter);
     PreparedQuery result = datastore.prepare(userQuery);
     Entity currentUser = result.asSingleEntity();
     return (String) currentUser.getProperty("username");
   }
 
   private boolean checkIfUserPreviouslyGuessedCorrect(String userId, EmbeddedEntity currentRound) {
-    EmbeddedEntity userGuessStatuses = (EmbeddedEntity) currentRound.getProperty("userGuessStatuses");
+    EmbeddedEntity userGuessStatuses =
+        (EmbeddedEntity) currentRound.getProperty("userGuessStatuses");
     boolean userGuessStatus = (Boolean) userGuessStatuses.getProperty(userId);
     return userGuessStatus;
   }
 
-  private void markUserGuessedCorrectly(String userId, EmbeddedEntity currentRound, Entity currentGame) {
-    EmbeddedEntity userGuessStatuses = (EmbeddedEntity) currentRound.getProperty("userGuessStatuses");
+  private void markUserGuessedCorrectlyAndAddPoints(
```

Comment:
I think we have an opportunity to add tests around the guess checking methods: checkIfUserPreviouslyGuessedCorrect(), markUserGuessedCorrectlyAndAddPoints(), isCorrectGuess() if you refactor them into their own class (e.g. GuessCheck, GuessArbiter, etc). The only issue is that markUserGuessedCorrectlyAndAddPoints() writes to Datastore, so you could either:
1. (Easiest) Have the method that calls markUserGuessedCorrectlyAndAddPoints() within ChatServlet do the writing to datastore
2. (Harder) Mock out the dependency to datastore in your tests for the new class

I think these tests will become even more important as we make the points logic more complicated, which is why I'm suggesting this. Take a little time to look into this and let me know if you have any questions.